app/Hutch.Relay/Services/TaskApiService.cs

```diff
@@ -0,0 +1,49 @@
+using System.Net;
+using Hutch.Rackit;
+using Hutch.Rackit.TaskApi.Contracts;
+using Hutch.Rackit.TaskApi.Models;
+using Hutch.Relay.Models;
+using Microsoft.Extensions.Options;
+
+namespace Hutch.Relay.Services;
+
+public class TaskApiService(
+  ILogger<TaskApiService> logger,
+  IOptions<ApiClientOptions> options,
+  ITaskApiClient upstreamTasks)
+{
+  public async Task SubmitResults(RelayTaskModel relayTask, JobResult jobResult
+  )
+  {
+    int retryCount = 0;
+    int delayInSeconds = 5;
+    int maxRetryCount = 5;
+    while (retryCount < maxRetryCount)
+    {
+      logger.LogInformation("Submitting Results..");
+      try
+      {
+        // Submit results upstream
+        var response = upstreamTasks.SubmitResultAsync(relayTask.Id, jobResult, options.Value);
+        if (response.IsCompletedSuccessfully)
+        {
+          logger.LogInformation("Successfully submitted results for {RelayTaskId}", relayTask.Id);
+          break;
+        }
+      }
```

Comment:
not seen this on a Task before.

But I'm a bit concerned about it as we don't currently await (per the above comment) so have we given it a chance to complete successfully?